### PR TITLE
Cm4370 write script to move a visualization from one host to another

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes CartoDB/cartodb-management#4370.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@javisantana @jgoizueta CR this, please.

**Comment by juanignaciosl** [comment]
@javisantana , @jgoizueta , ping ^

**Comment by javisantana** [comment]
I only miss the documenttion in operations -> https://github.com/CartoDB/cartodb/tree/master/doc/manual/source/operations

**Comment by juanignaciosl** [comment]
Sure, I'll document it as soon as it's merged :+1:

File: app/services/carto/visualizations_export_service.rb [hunk 1]
```diff
@@ -71,6 +75,39 @@ def import(visualization_id, skip_version_check = false)
...
+
+      add_data_layers(map, dump_data)
+
+      add_labels_layer(map, base_layer, dump_data)
+
+      set_map_data(map, dump_data)
+
+      description = dump_data["description"]
+
+      default_privacy = CartoDB::Visualization::Member::PRIVACY_LINK
```
Comment (jgoizueta) [inline app/services/carto/visualizations_export_service.rb]: Does the exported json (dump_data) have the privacy of the original visualization? if so wouldn't it be preferable to keep it? (if the new user can have it)

Comment (juanignaciosl) [inline app/services/carto/visualizations_export_service.rb]: Exported json hasn't it. Previous export code always set link privacy, but that privacy is not valid for some users (although the export/import use case is not likely to use standard account system).

**Comment by jgoizueta** [comment]
So when you want to import an exported visualization (json), you have to change the vis. id.

The owner id is replaced by the import process, but I see there are also uuids for layer ids and stat_tag. Do they need to be manually changed too?

**Comment by juanignaciosl** [comment]
@jgoizueta I should've given you some context.

This is not meant to be a 100% generic solution for all export/import use cases (although it's quite close) but a first step that covers this: CartoDB/cartodb-management#4370. I've just documented it as well, you'll find some additional explanation here: CartoDB/cartodb#6353.

Being a "visualization mover" is the reason why the user id is requested: we want to move a visualization among users, so the user must be changed. You don't need to change any id (including visualization one) if servers are different. In case of export/import inside the same server you must change all ids. I've documented it in the operations manual.

**Comment by jgoizueta** [comment]
:+1: OK everything seems fine to me

**Pull request merged**